### PR TITLE
Add TS implementation

### PR DESCRIPTION
**Pull request by ThatCopy**
To run
```bash
npm i
# rename the .env-template to .env and fill out your discord bot key and postges uri
npx prisma migrate dev --name init --preview-feature
npm run ts
npm run start
```

### CODE REVIEW

**Comment by stemount** [comment]
Very nice @ThatCopy. Can't fault this at all, I'll test it later 🥇

File: ts/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      user!.rep - 1
+  }
+}
+
+client.on('ready', () => {
+  console.log(`Logged in as ${client.user?.tag}!`)
+});
+
+client.on('message', async (msg) => {
+  const args: any[] = msg.content.trim().split(' ') // get args
```
Comment (Jaaneek) [inline ts/src/index.ts]: looks like string[] to me

File: ts/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  if (
+      msg.content.includes("++") || msg.content.includes("--") // check if the message contains -- or ++
+  ) {
+    let targetName: string = args.join(" ").replace(/(\+\+|--)\s*$/, "") // get the name
+    if(targetName.includes('@')) { // check if the name has a @ in it
+      const pingedUsed: Object = msg.mentions.users.toJSON() // get the mentioned user
```
Comment (Jaaneek) [inline ts/src/index.ts]: Possible typo pingedUsed > pingedUser
Dont you have a defined schema for this? Is this pingedUsed dynamic?

I think you should do interface PingedUser{

}

File: ts/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      const userId = Number(pingedUsed[0].id) // get the user id
```
Comment (Jaaneek) [inline ts/src/index.ts]: Will it always contain atleast one User?
If not you should probably check the length of pingedUser.

File: ts/src/index.ts (hunk 2 above)
Comment (Jaaneek) [reply ts/src/index.ts]: Also i would think of calling this variable pingedUsers since this is an array of Users right?

File: ts/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      await msg.channel.send(`${pingedUsed[0].username} has rep ${await repUpdate(userId, msg.content)}`)
+      return
+    }
+
+    let target = await msg.guild?.members.fetch() // get all members form the server
+    let user, userName 
+
+    for(const [key, e] of target!.entries()) {
+      if(e.nickname?.includes(targetName)) {user = e; userName = e.nickname; break} // if nickname matches the member break from loop
```
Comment (Jaaneek) [inline ts/src/index.ts]: I think adding comments above if is more readable

if(bool){ // some comment
}

// some comment
if(bool){ 
}

File: ts/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      if(e.user.username?.includes(targetName)) {user = e; userName = e.user.username; break} // if username matches the member break from loop
```
Comment (Jaaneek) [inline ts/src/index.ts]: I dont see the point to write this in one line.

just do 
if {

}
Much more readable

File: ts/src/index.ts (hunk 2 above)
Comment (ThatCopy) [reply ts/src/index.ts]: When I added a interface 
```ts
interface PingedUser {
  id: string,
  system?: any,
  locale?: any,
  flags: number,
  username: string,
  bot: boolean,
  discriminator: number,
  avatar?: string,
  lastMessageChannelID?: string,
  createdTimestamp: number,
  defaultAvatarURL?: string,
  tag: string,
  avatarURL?: string,
  displayAvatarURL?: string
}
```

The linter said this `TS2740: Type '{}' is missing the following properties from type 'PingedUser[]': length, pop, push, concat, and 26 more.`

File: ts/src/index.ts (hunk 5 above)
Comment (ThatCopy) [reply ts/src/index.ts]: I don't see how that is any more readable and I like having 3 short commands on one line, makes it look cleaner than if it's on 5 lines

Comment (Jaaneek) [reply ts/src/index.ts]: It might look cleaner but its less readable.

File: ts/src/index.ts (hunk 2 above)
Comment (Jaaneek) [reply ts/src/index.ts]: isnt toJSON() a way to tell an object how to parse it to a string? Basicly overwriting JSON.Stringify?

I think you should change that to JSON.parse(msg.mentions.users);

File: ts/src/index.ts (hunk 5 above)
Comment (stemount) [reply ts/src/index.ts]: @ThatCopy @Jaaneek we will m1 gang huddle and go through prettier/eslintrc and other linters and discuss this so we can find a balance between short handed stuff.

I'm a big fan of the 80 ruler so new line braces wins hands down so I side with @Jaaneek on this
```ts
if (this) {

}```

**Comment by stemount** [comment]
Totally made this go stale but actually @Jaaneek's feedback I will PR in for sure let's GOOO!

**Pull request closed without merging**